tests/scenario_tests/test_events_shared_channels.py

```diff
@@ -0,0 +1,514 @@
+import json
+from time import time, sleep
+
+from slack_sdk.signature import SignatureVerifier
+from slack_sdk.web import WebClient
+
+from slack_bolt import App, BoltRequest, Say
+from slack_bolt.authorization import AuthorizeResult
+from tests.mock_web_api_server import (
+    setup_mock_web_api_server,
+    cleanup_mock_web_api_server,
+)
+from tests.utils import remove_os_env_temporarily, restore_os_env
+
+valid_token = "xoxb-valid"
+
+
+def authorize(enterprise_id, team_id, client: WebClient):
+    assert enterprise_id == "E_INSTALLED"
+    assert team_id == "T_INSTALLED"
+    auth_test = client.auth_test(token=valid_token)
+    return AuthorizeResult.from_auth_test_response(
+        auth_test_response=auth_test, bot_token=valid_token,
+    )
+
+
+class TestEventsSharedChannels:
+    signing_secret = "secret"
+    mock_api_server_base_url = "http://localhost:8888"
+    signature_verifier = SignatureVerifier(signing_secret)
+    web_client: WebClient = WebClient(
+        token=None, base_url=mock_api_server_base_url,
+    )
+
+    def setup_method(self):
+        self.old_os_env = remove_os_env_temporarily()
+        setup_mock_web_api_server(self)
+
+    def teardown_method(self):
+        cleanup_mock_web_api_server(self)
+        restore_os_env(self.old_os_env)
+
+    def generate_signature(self, body: str, timestamp: str):
+        return self.signature_verifier.generate_signature(
+            body=body, timestamp=timestamp,
+        )
+
+    def build_headers(self, timestamp: str, body: str):
+        return {
+            "content-type": ["application/json"],
+            "x-slack-signature": [self.generate_signature(body, timestamp)],
+            "x-slack-request-timestamp": [timestamp],
+        }
+
+    valid_event_body = {
+        "token": "verification_token",
+        "team_id": "T_INSTALLED",
```

Comment:
for `app_mention` events, the top-level team_id is the one for an installed workspace.